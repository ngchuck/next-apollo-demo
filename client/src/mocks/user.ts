export const MOCK_USER = {
  firstName: 'Deep',
  lastName: 'Chakraborty',
    address: {
      streetName: '22 Royal crest Avenue'
    },
  age: 37,
  phone: '+44-7570324243',
  email: 'deep@gmail.com',
  avatar: '',
};