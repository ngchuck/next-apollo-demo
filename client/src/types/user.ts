export interface User {
  firstName: string;
  lastName: string;
  address: {
    streetName: string;
  };
  age: number;
  phone: string;
  email: string;
  avatar: string;
}