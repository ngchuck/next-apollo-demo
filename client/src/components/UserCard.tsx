import {User} from '@/types/user';
import styles from './UserCard.module.css'

const UserCard = ({user}: { user: User }) => (
  <div className={styles.card} key={user.email}>
    <h3>{user.firstName} {user.lastName} - {user.age}</h3>
    <p>{user.address.streetName}</p>
    <p>{user.email}</p>
    <p>{user.phone}</p>
    <img src={user.avatar}/>
  </div>
);
export default UserCard;
