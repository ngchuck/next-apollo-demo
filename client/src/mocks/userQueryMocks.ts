import {QUERY} from "@/components/User";


export const mocks = [
  {
    request: {
      query: QUERY,
      variables: {
        offset: 0,
        limit: 12
      },
    },
    result: {
      isLoading: false,
      data: {
        user: [{
          firstName: 'Deep',
          lastName: 'Chakraborty',
          address: {
            streetName: '22 Royal crest Avenue',
            __typename: 'address'
          },
          age: 37,
          phone: '+44-7570324243',
          email: 'deep@gmail.com',
          avatar: '',
          __typename: 'User'
        }],
      },
    },
  },
  {
    request: {
      query: QUERY,
      variables: {
        offset: 0,
        limit: 0
      },
    },
    result: {
      error: new Error("An error occurred")
    }
  },
  {
    request: {
      query: QUERY,
      variables: {
        offset: 0,
        limit: 1
      },
    },
    result: {
      isLoading: true
    }
  }
];