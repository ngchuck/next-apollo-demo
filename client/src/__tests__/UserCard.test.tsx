import { render } from '@testing-library/react'
import UserCard from "@/components/UserCard";
import { MOCK_USER } from "@/mocks/user";


it('renders UserCard unchanged', () => {
  const { container } = render(<UserCard user={MOCK_USER}/>)
  expect(container).toMatchSnapshot()
});