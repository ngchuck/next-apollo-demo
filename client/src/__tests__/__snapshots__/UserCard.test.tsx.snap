// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UserCard unchanged 1`] = `
<div>
  <div
    class="card"
  >
    <h3>
      Deep
       
      Chakraborty
       - 
      37
    </h3>
    <p>
      22 Royal crest Avenue
    </p>
    <p>
      deep@gmail.com
    </p>
    <p>
      +44-7570324243
    </p>
    <img
      src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/716.jpg"
    />
  </div>
</div>
`;
